### PR TITLE
Vendor HID

### PR DESCRIPTION
**Pull request by kaczmarczyck**
Introduces a second HID parser. In this PR still, the `CtapState` treats packets from both HIDs indiscriminately. So from here on, when we introduce the second HID page to the driver and use it in main, we should start seeing results.

The next PR will focus on processing the incoming channels differently. Afterwards, all that is left to do to actually make it useful is outside of the CTAP library.

### CODE REVIEW

**Comment by kaczmarczyck** [comment]
I also updated the workflows to add the new feature. It is worth noting that binary size increases by ~100B with the feature, and does not increase without the feature.

File: Cargo.toml [hunk 1]
```diff
@@ -31,6 +31,7 @@ std = ["crypto/std", "lang_items/std", "persistent_store/std"]
 verbose = ["debug_ctap", "libtock_drivers/verbose_usb"]
 with_ctap1 = ["crypto/with_ctap1"]
 with_nfc = ["libtock_drivers/with_nfc"]
+with_vendor_hid = []
```
Comment (ia0) [inline Cargo.toml]: Let's not propagate this [bad style](https://rust-lang.github.io/api-guidelines/naming.html#feature-names-are-free-of-placeholder-words-c-feature) and name the feature `vendor_hid`.

File: src/lib.rs [hunk 2]
```diff
@@ -87,6 +99,12 @@ impl<E: Env> Ctap<E> {
                 self.hid
                     .process_hid_packet(&mut self.env, packet, now, &mut self.state)
             }
+            #[cfg(feature = "with_vendor_hid")]
+            Transport::VendorHid => {
+                self.vendor_hid
+                    .process_hid_packet(&mut self.env, packet, now, &mut self.state)
+            }
+            #[cfg(not(feature = "with_vendor_hid"))]
             Transport::VendorHid => HidPacketIterator::none(),
```
Comment (ia0) [inline src/lib.rs]: If we protect the code with a feature, shouldn't we also protect the transport enum value? It looks weird to have that enum value if it's not really used.

**Review by ia0 — DISMISSED** [review]

File: src/lib.rs (hunk 2 above)
Comment (kaczmarczyck) [reply src/lib.rs]: Done

File: Cargo.toml (hunk 1 above)
Comment (kaczmarczyck) [reply Cargo.toml]: I only renamed the new feature. Also added the flag for deploy, even though it's not useful yet.

**Review by ia0 — APPROVED** [review]

**Pull request merged**